packages/core/src/utils/get-packages.ts

```diff
@@ -52,14 +50,19 @@ interface GetPackageEntryPointsOpts {
 export const getPackageEntryPoints = async ({
   packageRoot,
 }: GetPackageEntryPointsOpts): Promise<PackageEntryPoint[]> => {
-  const entryPaths = await glob(['src/index.ts', 'src/entries/*.ts'], {
+  const defaultEntry = 'src/index.ts';
+  const entryPaths = await glob([defaultEntry, 'src/entries/**/*.ts'], {
     cwd: packageRoot,
     absolute: true,
   });
 
   return entryPaths.map((entryPath) => {
-    const isDefaultEntry = entryPath.includes('src/index.ts');
-    const entryName = isDefaultEntry ? 'dist' : basename(entryPath);
+    const isDefaultEntry = entryPath.includes(defaultEntry);
+    const entryName = isDefaultEntry
+      ? 'dist'
+      : path
+          .relative(path.join(packageRoot, 'src/entries'), entryPath)
+          .replace(/\..*/, '');
```

Comment:
Might be worth adding `$` to this regex so it will only ever replace `.*` from the end of the string.
```suggestion
          .replace(/\..*$/, '');
```